presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -0,0 +1,380 @@
+    {
+        return PreparedStatement::setBoolean;
+    }
+
+    public static ColumnMapping tinyintColumnMapping()
+    {
+        return ColumnMapping.longMapping(TINYINT, ResultSet::getByte, tinyintWriteFunction());
+    }
+
+    public static LongWriteFunction tinyintWriteFunction()
+    {
+        return (statement, index, value) -> statement.setByte(index, SignedBytes.checkedCast(value));
+    }
+
+    public static ColumnMapping smallintColumnMapping()
+    {
+        return ColumnMapping.longMapping(SMALLINT, ResultSet::getShort, smallintWriteFunction());
+    }
+
+    public static LongWriteFunction smallintWriteFunction()
+    {
+        return (statement, index, value) -> statement.setShort(index, Shorts.checkedCast(value));
+    }
+
+    public static ColumnMapping integerColumnMapping()
+    {
+        return ColumnMapping.longMapping(INTEGER, ResultSet::getInt, integerWriteFunction());
+    }
+
+    public static LongWriteFunction integerWriteFunction()
+    {
+        return (statement, index, value) -> statement.setInt(index, toIntExact(value));
+    }
+
+    public static ColumnMapping bigintColumnMapping()
+    {
+        return ColumnMapping.longMapping(BIGINT, ResultSet::getLong, bigintWriteFunction());
+    }
+
+    public static LongWriteFunction bigintWriteFunction()
+    {
+        return PreparedStatement::setLong;
+    }
+
+    public static ColumnMapping realColumnMapping()
+    {
+        return ColumnMapping.longMapping(REAL, (resultSet, columnIndex) -> floatToRawIntBits(resultSet.getFloat(columnIndex)), realWriteFunction());
+    }
+
+    public static LongWriteFunction realWriteFunction()
+    {
+        return (statement, index, value) -> statement.setFloat(index, intBitsToFloat(toIntExact(value)));
+    }
+
+    public static ColumnMapping doubleColumnMapping()
+    {
+        return ColumnMapping.doubleMapping(DOUBLE, ResultSet::getDouble, doubleWriteFunction());
+    }
+
+    public static DoubleWriteFunction doubleWriteFunction()
+    {
+        return PreparedStatement::setDouble;
+    }
+
+    public static ColumnMapping decimalColumnMapping(DecimalType decimalType)
+    {
+        // JDBC driver can return BigDecimal with lower scale than column's scale when there are trailing zeroes
+        int scale = decimalType.getScale();
+        if (decimalType.isShort()) {
+            return ColumnMapping.longMapping(decimalType,
+                    (resultSet, columnIndex) -> encodeShortScaledValue(resultSet.getBigDecimal(columnIndex), scale),
+                    shortDecimalWriteFunction(decimalType))
+                    .withPredicatePushdownFilter(domain -> false); // TODO enable
+        }
+        return ColumnMapping.sliceMapping(
+                decimalType,
+                (resultSet, columnIndex) -> encodeScaledValue(resultSet.getBigDecimal(columnIndex), scale),
+                longDecimalWriteFunction(decimalType))
+                .withPredicatePushdownFilter(domain -> false); // TODO enable
```

Comment:
Decimal predicate pushdown was not allowed in `io.prestosql.plugin.jdbc.QueryBuilder#isAcceptedType` and this is why it's retained as not allowed in this refactoring.